### ISSUE TITLE
Screencapture performance

### ISSUE THREAD

**Issue by paulirish** [openstf/stf#11]
It's crazy good. seriously. Kicks the pants off Chrome devtools screencasting. :)

That said there appears to be something blocking a solid 60fps.

Here's what i'm seeing after profiling the full STF app using about:tracing

### one frame of work
![image](https://cloud.githubusercontent.com/assets/39191/8643213/c664bf52-28e5-11e5-80c2-239c8d04b414.png)

the real thing here is that we're in the resourcefetcher for the blob for a long ass time. that's really what's slowing down this process. 

Here's the start: https://code.google.com/p/chromium/codesearch#chromium/src/third_party/WebKit/Source/core/fetch/ResourceFetcher.cpp&q=%22%20TRACE_EVENT_ASYNC_BEGIN2(%22blink.net%22,%20%22Resource%22%22&sq=package:chromium&type=cs&l=901
Here's the end: https://code.google.com/p/chromium/codesearch#chromium/src/third_party/WebKit/Source/core/fetch/ResourceFetcher.cpp&q=%22TRACE_EVENT_ASYNC_END0(%22blink.net%22,%20%22Resource%22%22&sq=package:chromium&type=cs&l=797

what's interesting to me is that if i poke inside of the browser process... it reports finishing loading the blob at about halfway: 
![image](https://cloud.githubusercontent.com/assets/39191/8643444/c35e7304-28e8-11e5-8a74-3b2cb511331c.png)
(the actual "requestcomplete" on the browserthread is offscreen in the screenshot)

it's possible whatever is handling the blob URL is unimplemented, but right now it appears to be taking longer than I'd expect. might be a chrome bug.

### sometimes I see a missed frame
![image](https://cloud.githubusercontent.com/assets/39191/8643223/f3087d96-28e5-11e5-8c32-efa8b936bb7b.png)

In these cases we do two image decodes and throw one of them away immediately. 
Not sure yet why this is happening.

<hr>

Last thing, I'd consider trying out requestAnimationFrame, only because it provides the scheduler with a lot more information on when you're ready to paint a frame. It has a low-latency mode so you can call rAF immediately once the image has loaded and you should get an instant paint.

<hr>
Here's the trace file i'm looking at: https://dl.dropboxusercontent.com/u/39519/temp/trace_stf-screencapture.json.gz

**Comment by gunta** [comment]
Thank you!
We'll have a look.

**Comment by sorccu** [comment]
Truth to be told we have spent relatively little time perfecting the client-side implementation (which is actually quite hacky), because we initially were not able to produce frames quickly enough to start with. And even though we now are able to do that, we still had our old assumptions and didn't even bother checking if there might be some additional gains to be made there. Thanks for the reminder and detailed information!

**Comment by paulirish** [comment]
yeah no worries and no rush. i was just curious. nice to see you now can generate frames fast enough that the browser has to catch up. a good problem to have!

**Comment by paulirish** [comment]
cc @dmurph as this appears to be a decent real-world stresstest  for blob performance in Chrome.

**Comment by nanoscopic** [comment]
Why would you want to stream 60fps of jpegs to a browser? That seems like a poor idea. Is there some goal to this ticket?

If you want to see 60fps then streaming h264 should be used.

I think this ticket should be closed; it doesn't have a clear goal or way to complete.
